### DISCUSSION TITLE
How to get full file url in the backend

### DISCUSSION THREAD

**Discussion by molotow11** [keystonejs/keystone#8385] [Questions] [answered]
Hello, Guys!

From the backend `db.findOne` function we have only `file_filename`, how to get full path url like from GraphQL response?

**Comment by arsekil** [comment]
Does it have perhaps a `url` property to use?

**Comment by marekryb** [comment, marked as answer]
Hi,

One way would be to use ```query.findOne``` API instead since you can get GraphQL structures this way (including url).
However if you want to use ```db.findOne``` then you should perform same transformation as ```generateUrl``` that you already have specified in [storage configuration](https://keystonejs.com/docs/config/config#storage-images-and-files)
